src/object.d

```diff
@@ -1431,6 +1431,132 @@ class TypeInfo_Struct : TypeInfo
     assert(!typeid(S).equals(&s, &s));
 }
 
+class RTTypeid(T) : TypeInfo
+if (is(T == struct))
+{
+    override string toString() const { return T.stringof; }
+
+    override bool opEquals(Object rhs)
+    {
+        if (this is rhs)
+            return true;
+        auto s = cast(const RTTypeid!T) rhs;
+        return s !is null;
+    }
+
+    override size_t getHash(scope const void* p) @trusted pure nothrow const
+    {
+        assert(p);
+        // Canonical prototype
+        alias HashFun = extern(D) size_t delegate() const nothrow @safe;
+        enum bool hasHash = is(typeof(&(cast(S*) null).toHash) == HashFun);
```

Comment:
`nothrow` and `@safe` are recommended but not enforced:
https://run.dlang.io/is/ZpYMqt
```suggestion
        alias HashFun = extern(D) size_t delegate() const;
        enum bool hasHash = is(typeof(&(cast(S*) null).toHash) : HashFun);
```

and add a try-catch if necessary below.